go/arrow/array/numeric.gen.go

```diff
@@ -575,10 +588,21 @@ func (a *Float32) GetOneForMarshal(i int) interface{} {
 func (a *Float32) MarshalJSON() ([]byte, error) {
 	vals := make([]interface{}, a.Len())
 	for i := 0; i < a.Len(); i++ {
-		if a.IsValid(i) {
-			vals[i] = a.values[i]
-		} else {
+		if !a.IsValid(i) {
 			vals[i] = nil
+			continue
+		}
+
+		f := a.Value(i)
+		switch {
+		case strconv.FormatFloat(float64(f), 'g', -1, 32) == "NaN":
+            vals[i] = "NaN"
+		case strconv.FormatFloat(float64(f), 'g', -1, 32) == "+Inf":
+            vals[i] = "+Inf"
+		case strconv.FormatFloat(float64(f), 'g', -1, 32) == "-Inf":
+            vals[i] = "-Inf"
+		default:
+			vals[i] = f
 		}
```

Comment:
Please fix the indentation.

Also, let's only call `strconv.FormatFloat` once: 

```go
switch v := strconv.FormatFloat(float64(f), 'g', -1, 32) {
case "NaN", "+Inf", "-Inf":
  vals[i] = v
default:
  vals[i] = f
}
```